### ISSUE TITLE
use Trello for backlog

### ISSUE THREAD

**Issue by tobibeer** [Jermolene/TiddlyWiki5#1466]
@Jermolene, while GitHub's issue mgt is quite great it gives little features for actually managing a backlog, prioritizing, categorizing, basic Kanban, etc...

Perhaps consider using a trello to manage what's up next on the prio / sprint / debugging / documentation / feature list? I think the overhead is worth it. So, one can be used to plan / track progress and the other to track issues.

I just stumbled over the [brackets trello board](https://trello.com/b/LCDud1Nd/brackets
) and I think there are a few things worth considering...

* offical backlog
* community backlog
* "Nominations" — voting
    * you could poll on "what is your preference for the next release"
* in progress
* done (release x)

...one thing is obvious, this is their public board and there are other boards being managed in the background from which they move items back and forth. So, there could be a releases board, to which the finished release ist can be moved and then a new one created for the next release, etc...

**Comment by Jermolene** [comment]
Hi @tobibeer I've used Trello myself a little to manage my own tasks, it's very cool. Is there a way of syncing GitHub issues with it?

**Comment by tobibeer** [comment]
mhhh, not sure ...perhaps https://waffle.io looks more interesting in terms of automated syncing — free for public repos, not sure if it can cope with trello

**Comment by pmario** [comment]
There is an Assignee field here on github but we don't use it. .. It's also not clear, if a user, who is willing to assign himself to work on an issue, is able to do so. .. I think we need a tiddlywiki organisation, with different roles and not a new 3rd party tool.

**Comment by tobibeer** [comment]
To me, assigning isn't quite the most important thing... it's really giving that backlog some ordering and perhaps polling preferences.

**Comment by Jermolene** [comment]
>  I think we need a tiddlywiki organisation, with different roles and not a new 3rd party tool.

Gulp. We could transfer the TW5 repo to github.com/TiddlyWiki, and GitHub would do the necessary redirects.

**Comment by pmario** [comment]
Voting -> Everyone can add a +1 post here, if she has a github account.
Backlog -> activate the "Label: bug" filter in the issue overview.  

We have 9 different labels:
  bug, discussion, documentation, improvement, needs-work, newfeature, refactor, needs-cla-signature, wontfix

If a contributor can set a label: work-in-progress  and an Assignee, we get a simple "in progress" list.

Everyone, that wants to contribute to the docs, can get a list of open issues with the "documentation" filter

needs-work is for pull requests, that need improvements. 

If Jeremy wants to set up a poll, it would be possible to tag a "newfeature" with "poll" and users would be able to find open polls. 

If a "newfeature" is marked with a Milestone, we know what's planned. .... 

IMO the possibilities are there, we need to use them. ... but imo if Jeremy is the only one, who can do this, managing it, is a full time job. .... But the same is true if we move over to trello. I like, how trello looks like bit imo adding trello just moves the problem and additional work to a different platform. 

just my 2 cents.

**Comment by tobibeer** [comment]
>  Everyone can add a "+1 plus some reasoning" post here, if she has a github account.

Yes, but that's not polling.

**Comment by pmario** [comment]
IMO Milestones tend to be broken. ... eg: issues, that are assigned to M1 can't be fixed in time, so they are moved to M2, then M3, M4 .... There have been a lot of those in the old TWc subversion bug tracker.

But Milestones could be used to implement the [Eisenhower Principle](http://en.wikipedia.org/wiki/Time_management#The_Eisenhower_Method). It's simple and battle tested.

 - urgent-important
    - Jeremy needs to deal with it, now.
       - eg: critical bugs

 - not-urgent-important 
    - should be delegated. 
        - eg: documentation improvements, community communications, ...

 - urgent-not-important 
    - must be delegated and should to be reviewed.
    - or must be delegated and should be terminated.
       - eg: community support requests the land here, but could be at the user group.

 - not-urgent-not-important
    - eg: prepare for wontfix

------------------

Since some of those labels are not polite, if you need to classify issues. 
eg: community support request that lands here -> "urgent but not important" ... this Milestone text should be "urgent" ... If a user reads "urgent" ... community support ... imo this is polite. So let's reformat it a bit. 

 - Priority One   (urgent-important)
    - Jeremy needs to deal with it, now.
       - eg: critical bugs

 - Important    (not-urgent-important)
    - should be delegated 
        - eg: documentation improvements, community communications, ...

 - Urgent    (urgent-not-important )
    - must be delegated and should to be reviewed.
       - eg: community support requests the land here, but could be at the user group.

 - wontfix    (not-urgent-not-important)   
    - eg: prepare for wontfix -> is no Milestone it is a label for a closed issue

---------------

The only task to do is, that we need to classify the issues.

what do you think?

**Comment by tobibeer** [comment]
> IMO Milestones tend to be broken.

I wasn't suggesting any milestones, just a basic, manually ordered backlog of all issues. Perhaps with a simple marker where Jeremy could set the scope for the next release, i.e. in or out.

Simple drag and drop, done. Tags are just on top for issue management, but not necessarily the deciding element for managing the backlog... except perhaps for having different, individually prioritized lists and some things being more important than others, e.g. bugs > doc.

I sure like the Eisenhower principle, but then this is implicit for me in how Jeremy decides to eventually order that list or those lists ...importance would be something that user's might indicate via poll results, e.g. sorted by number of votes, rather than looking at issues and counting some icons. ^^

**Comment by pmario** [comment]
>>IMO Milestones tend to be broken.

>I wasn't suggesting any milestones, ...

I was referring to my post and the Milestone info there.

**Comment by pmario** [comment]
> I sure like the Eisenhower principle, but then this is implicit for me in how Jeremy decides to eventually order that list or those lists ...importance would be something that user's might indicate via poll results, e.g. sorted by number of votes, rather than looking at issues and counting some icons. 

Voting is nice, but not necessary valid for community contributions. ... It's nice that a feature request has a lot of votes. .... But as a contributor I would choose a feature, that I'm capable of solving and that is of interest for me. .. So I'd choose to create a pull request for a feature that I want to implement.

If votes have some bitcoins attached, that would make a difference ;)

**Comment by tobibeer** [comment]
Yes, both makes sense. Surely, contributors pick up the issues that scratch their itch... even though that's perhaps not the most preferable way. I can imagine with a plain, sorted backlog of all issues, even intermingled there would be more focus on what's prioritized... even if only tendentially. Right now, that's a little too obscure.

**Comment by Jermolene** [comment]
To be honest, I'm not a massive fan of using the assignee field of an issue as a way to "claim" it. Another way to claim the issue is to submit a pull request that references it. At least that way one can see whether the would-be claimer is competent to claim the issue. If somebody wants to embark on a big piece of work to fix a bug it's probably more important that they talk to the community for feedback.

And as for voting, I don't think it has much place. I'm not against occasional polls to help canvas opinions, but I think the idea of being able to vote on any bug isn't very helpful. I'd rather read a message from the user explaining why they want the feature, what makes it important to them.

Basically, these mechanisms are kind of "free" for the person pushing the button to claim a bug or vote for it. That reduces the value of their feedback. If somebody submits a garbled horrible fix for a bug, even if I can't use it I know that that user is really serious about needing to get the bug fixed. It's a much more useful signal than them having voted on something.

**Comment by tobibeer** [comment]
>  It's a much more useful signal than them having voted on something.

There are not only bugs. There are also things "in the pipe" in terms of "new features". While we can have long debates over why and what there is a good indicator to a vote button as to how much people think something is important. Of course, that implies people actually find and use that button.

**Comment by nicholas-spies** [comment]
I, too, think that voting is hardly the way to make decisions about
software, its implementation, its potential uses, etc. This has to issue
from a strong vision, such as one that has been provided by Jeremy, and
amplified by a relatively small group who have shared Jeremy's vision  in
his invention of TiddlyWiki to begin with. At the same time, these
principles should be pounded out as a series of specifications that can
serve as a guide for comment, by seasoned developers and by newcomers,
addressing the same core vision, so as to understand it well enough to make
meaningful, suggestions for modifications of various features, and so
forth.

Something as complicated as a many-faceted program that has aleady proven
to have many unforeseen applications, must have, at its core, a strongly
worded statement of its objectives, the sorts of tasks that might find
application using it, but set in a manner that does not jeopardize the very
strengths that it has proven to possess.

So, in a way, the main task of TiddlyWiki project is to formulate, realize,
and implement a new paradigm, which in many ways it already has. In this
instance, the paradigm happens to have been implemented in JavaScript, but
it could, in principle, be realized in some other language provided the
reasons to do so were compelling enough, or that this would lead to opening
news avenues for application.

Best,

--Nick Spies

**Comment by tobibeer** [comment]
I guess this can be closed. (Sorry for my absence).

**Comment by Jermolene** [comment]
Hey @tobibeer good to see you back. Hopefully you'll see a few welcome changes

**Comment by tobibeer** [comment]
Will take a look. :+1:
